test/libsolidity/semanticTests/inlineAssembly/tstore_forbidden_staticcall.sol

```diff
@@ -0,0 +1,18 @@
+contract D {
+    function f() external returns (uint x) {
+        assembly {
+            tstore(0, 42)
+        }
+    }
+}
+
+contract C {
+    function test() external returns (uint y) {
+        D d = new D();
+        address(d).staticcall(abi.encodeWithSignature("f()"));
+    }
+}
+// ====
+// EVMVersion: >=cancun
+// ----
+// test() -> 0
```

Comment:
This should revert.

Reply:
An additional test you can add is this rather pathological construction:
```
contract C {
    function f() internal {
        assembly { tstore(0, 0) }
    }
    function g() public view {
        function() internal ptr = f;
        function() internal view ptr2;
        assembly { ptr2 := ptr }
        ptr2(); // we force calling the non-view function, which should result in a revert during the staticcall
    }
    function test() public {
        this.g(); // an external call to a view function should use static call
    }
}
// ----
// test() -> FAILURE
```

This should actually show the revert itself - it's using assembly to pretend that a non-view function is view and callable in the view function ``g``. When calling ``this.g()``, this should result in a ``STATICCALL`` (since ``g`` is ``view``) - but then revert, since it does still call ``f`` using the assembly trickery.